src/pages/auth/Login/Login.js

```diff
@@ -0,0 +1,128 @@
+// LoginForm.js
+
+import React, { useEffect, useRef, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import * as Yup from 'yup';
+import { Link } from 'react-router-dom';
+import { AiFillEye, AiFillEyeInvisible } from 'react-icons/ai';
+import { loginThunk } from '../../../slices/authSlice/authSlice';
+import { setNotice, setAlert, clearAll } from '../../../slices/appSlice/appSlice';
+import ErrorMessage from '../../../components/messages/ErrorMessage';
+
+const LoginSchema = Yup.object().shape({
+  email: Yup.string().email('Invalid email').required('Email is required!!'),
+  password: Yup.string().min(6, 'Password is Too Short!').max(50, 'Password is Too Long!').required('Email is Required'),
+});
+
+const Login = () => {
+  const dispatch = useDispatch();
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [errors, setErrors] = useState({});
+  const [showPassword, setShowPassword] = useState(false);
+  const passwordRef = useRef(null);
+  // handle app state
+
+  useEffect(() => (() => {
+    dispatch(clearAll());
+  }), [dispatch]);
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    // validate the form
+    LoginSchema.validate({ email, password }, { abortEarly: false }).then((res) => {
+      // if validation passes, dispatch the thunk
+      dispatch(loginThunk(res)).then((res) => {
+        if (res.error) {
+          // if the thunk fails, set the errors
+          dispatch(setAlert(res.payload));
+        } else {
+          // if the thunk succeeds, set the notice
+          dispatch(setNotice('Login successful'));
+        }
+      });
+    }).catch((err) => {
+      // if validation fails, set the errors
+      const newErrors = {};
+      err.inner.forEach((error) => {
+        newErrors[error.path] = error.message;
+      });
+      setErrors(newErrors);
+    });
+  };
+
+  const showHandler = (e) => {
+    e.preventDefault();
+    setShowPassword(!showPassword);
+    if (passwordRef.current.type === 'password') {
+      passwordRef.current.type = 'text';
+    } else {
+      passwordRef.current.type = 'password';
+    }
+  };
+
+  return (
+    <section className="bg-gray-50 dark:bg-gray-900">
```

Comment:
- [ ] Please make sure to fix the height of this section to shows up correctelly on mobile screens

![mobile-bg](https://user-images.githubusercontent.com/95412943/230319812-0184b394-93f5-4710-a2ea-86216f99b0f9.png)